### PR TITLE
feat(plugins) support plugin ordering

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:
Yet more PRs because GitHub Actions got sad and didn't want to run on the old one anymore.

Adds support for Plugin Ordering.

Adds integration tests against nightly Kong images. Currently omits nightly tests of OSS images because [they're not ready yet](https://github.com/Kong/kubernetes-ingress-controller/pull/2657#issuecomment-1180699982). I don't have any tracking information from the enablement team about when those tags will be available, unfortunately.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: Fix https://github.com/Kong/kubernetes-ingress-controller/issues/2656

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR
](https://github.com/Kong/kubernetes-ingress-controller/pull/2657)%5D(https://github.com/Kong/kubernetes-ingress-controller/pull/2657)

### CODE REVIEW

File: .github/workflows/test.yaml
Comment (shaneutt) [inline .github/workflows/test.yaml:254]: Is this disable something we need to undo before we merge? if not, we should add a TODO and link an issue.

File: .github/workflows/test.yaml
Comment (shaneutt) [inline .github/workflows/test.yaml:320]: I recommend we either uncomment OR add a TODO and link a follow up issue depending on the context.

File: test/integration/suite_test.go
Comment (shaneutt) [inline test/integration/suite_test.go:65]: technically you're not catching the inverse, where a password or a tag is provided but not the other. You could solve this by doing:

```golang
if (cond1 || cond2) && !(cond1 && cond2) {
    exitOnErrWithCode(fmt.Errorf("username AND password must be provided"))
```

**Review by shaneutt — CHANGES_REQUESTED** [review]
Looks good, only minor comments to resolve before merge :+1:

File: .github/workflows/test.yaml
Comment (pmalek) [inline .github/workflows/test.yaml:177]: This is not set in the repo/env secrets anymore after https://github.com/Kong/kubernetes-ingress-controller/pull/2648 got merged. If you take a look at the changes it introduced it's gonna be fairly easy to deduce what step to add here to pull the license in.

**Review by pmalek — CHANGES_REQUESTED** [review]

File: test/integration/plugin_test.go
Comment (pmalek) [reply test/integration/plugin_test.go:265]: I had to have made this comment half unconscious. This comment doesn't belong to this line but to lines actually using `assert`/`require` helpers.

My main concern with those are situations like these:

```go
n, err := b.ReadFrom(resp.Body)
require.NoError(t, err)
```

Is this an error that should warrant the whole test to fail? It might be a consequence of lack of free memory on a box that this test was running on. Should that fail the test? It's not production code: that's test code that does **additional** actions which wouldn't occur in production environments.

> If you're not persuaded, I would ask that we at least not block this PR for this purpose since this is not the only place this is being done, and instead discuss this further in a forum that wont hold up this work (github discussions, slack, zoom, e.t.c.) and if we come to a consensus that we do want to remove these then remove them holistically.

No intention to block this PR on this matter. 👍 I'll bring this topic up in a separate discussion

Comment (shaneutt) [reply test/integration/plugin_test.go:265]: Alright I don't want to overload this one PR I'll follow up in #2629  :+1:

**Comment by rainest** [comment]
GitHub's way of telling you (or rather, not telling you, since it just says "startup failure" and nothing else) that "an action can't run because it needs a step that doesn't exist" is very bad :(

**Review by pmalek — COMMENTED** [review]
Minor nits, overall 👍

**Review by shaneutt — APPROVED** [review]
Once @pmalek's is :+1: I'm :+1:

File: .github/workflows/test.yaml
Comment (rainest) [reply .github/workflows/test.yaml:182]: It's the internal version of https://hub.docker.com/r/kong/kong-gateway/, i.e. the image that contains Enterprise stuff. Our permissions on Docker Hub usually only extend access to internal resources to various bot users, so it looks like a 404 on a  personal account.

File: test/integration/suite_test.go
Comment (rainest) [reply test/integration/suite_test.go:65]: What would prompt us to do so? The new code here is similar to the surrounding code; it's all conditionals that maybe add something additional to the Kong plugin builder.

File: test/integration/plugin_test.go
Comment (rainest) [reply test/integration/plugin_test.go:305]: As with most uses of `Eventually()` I wish there was a way to do error handling at the end if the condition never succeeded. In practice we don't expect this to fail indefinitely ever, and are abusing it as a retry function in case there's some transient modification to the resource and we need to `Get()` again, and the logs would usually just be noise telling us that we did indeed retry.

File: .github/workflows/test.yaml
Comment (pmalek) [reply .github/workflows/test.yaml:182]: Ok got it, in that case perhaps we could add some documentation around it (I don't care if it's going to be here or elsewhere in the repo)?

I'm just trying to prevent folks - reading this in the future - scratching their heads, trying to figure out what's that.

File: test/integration/suite_test.go
Comment (pmalek) [reply test/integration/suite_test.go:65]: > What would prompt us to do so?

The fact that it's growing into a giant if/else `func`. Slapping a name on it and extracting it to a separate func would certainly help decrease the mental burden when reading this don't you think?

File: test/integration/plugin_test.go
Comment (pmalek) [reply test/integration/plugin_test.go:305]: The reason I suggested this is that without the log we'll never know what was the error.

File: test/integration/suite_test.go
Comment (pmalek) [inline test/integration/suite_test.go:36]: Without a named return variable it would be ideal to add a comment what this function returns.

File: test/integration/suite_test.go
Comment (pmalek) [inline test/integration/suite_test.go:77]: Nice 🥇

**Review by pmalek — COMMENTED** [review]
One minor thing and we're good to go from my side

**Review by pmalek — APPROVED** [review]

**Comment by pmalek** [comment]
I guess you'll need to make linter happy though

**Review by pmalek — APPROVED** [review]

**Review by pmalek — APPROVED** [review]

**Pull request merged**
